src/framework/components/anim/system.js

```diff
@@ -50,7 +51,16 @@ class AnimComponentSystem extends ComponentSystem {
             data.layers.forEach((layer, i) => {
                 layer._controller.states.forEach((stateKey) => {
                     layer._controller._states[stateKey]._animationList.forEach((node) => {
-                        component.layers[i].assignAnimation(node.name, node.animTrack);
+                        if (!node.animTrack || node.animTrack === AnimTrack.EMPTY) {
+                            const animationAsset = pc.app.assets.get(layer._component._animationAssets[layer.name + ':' + node.name].asset);
+                            if (animationAsset && !animationAsset.loaded) {
```

Comment:
What if the animation asset is loaded for some reason? Should we be calling `component.layers[i].assignAnimation(node.name, animationAsset.resource);` or will that be happening anyway on the load?

Reply:
If the animationAsset has been loaded the node will already contain a valid animTrack so this wont be reached.

Reply:
Might be worth adding a comment on why we don't have the other path in the code